app/actions/websocket/system.ts

```diff
@@ -30,12 +32,30 @@ export async function handleConfigChangedEvent(serverUrl: string, msg: WebSocket
 
     try {
         const config = msg.data.config;
-        const systems: IdValue[] = [{
-            id: SYSTEM_IDENTIFIERS.CONFIG,
-            value: JSON.stringify(config),
-        }];
+        const systems: IdValue[] = [{id: SYSTEM_IDENTIFIERS.CONFIG, value: JSON.stringify(config)}];
+
+        const prevConfig = await getConfig(operator.database);
         await operator.handleSystem({systems, prepareRecordsOnly: false});
+        checkDisplayNameSettings(serverUrl, config, prevConfig);
```

Comment:
here wrap in an if statement that checks if the LockTeammateNameDisplay property changed between the new config and the previous one

Prefer the example above